nilearn/connectome/tests/test_connectivity_matrices.py

```diff
@@ -259,6 +187,192 @@ def random_non_singular(p, sing_min=1.0, sing_max=2.0, random_state=0):
     return unitary1.dot(diag).dot(unitary2.T)
 
 
+def _assert_connectivity_tangent(connectivities, conn_measure, covs):
+    for k, cov_new in enumerate(connectivities):
+        # Positive definiteness if expected and output value checks
+
+        assert_array_almost_equal(cov_new, cov_new.T)
+
+        gmean_sqrt = _map_eigenvalues(np.sqrt, conn_measure.mean_)
+
+        assert is_spd(gmean_sqrt, decimal=7)
+        assert is_spd(conn_measure.whitening_, decimal=7)
+        assert_array_almost_equal(
+            conn_measure.whitening_.dot(gmean_sqrt),
+            np.eye(N_FEATURES),
+        )
+        assert_array_almost_equal(
+            gmean_sqrt.dot(_map_eigenvalues(np.exp, cov_new)).dot(gmean_sqrt),
+            covs[k],
+        )
+
+
+def _assert_connectivity_precision(connectivities, covs):
+    for k, cov_new in enumerate(connectivities):
+        assert is_spd(cov_new, decimal=7)
+        assert_array_almost_equal(cov_new.dot(covs[k]), np.eye(N_FEATURES))
+
+
+def _assert_connectivity_correlation(connectivities, cov_estimator, covs):
+    for k, cov_new in enumerate(connectivities):
+        assert is_spd(cov_new, decimal=7)
+
+        d = np.sqrt(np.diag(np.diag(covs[k])))
+
+        if cov_estimator == EmpiricalCovariance():
+            assert_array_almost_equal(d.dot(cov_new).dot(d), covs[k])
+        assert_array_almost_equal(np.diag(cov_new), np.ones(N_FEATURES))
+
+
+def _assert_connectivity_partial_correlation(connectivities, covs):
+    for k, cov_new in enumerate(connectivities):
```

Comment:
We don't need the index : 

```suggestion
    for cov, cov_new in zip(covs, connectivities):
```